content/tokio/tutorial/bridging.md

```diff
@@ -0,0 +1,384 @@
+|
+| This means that when a call to `block_on` on the `Runtime` returns, all
+| spawned tasks on that runtime will freeze until you call `block_on` on the
+| runtime again.
+
+Once we have this struct, most of the methods are easy to implement:
+```rs
+use bytes::Bytes;
+use std::time::Duration;
+
+impl Client {
+    pub fn get(&mut self, key: &str) -> crate::Result<Option<Bytes>> {
+        self.rt.block_on(self.inner.get(key))
+    }
+
+    pub fn set(&mut self, key: &str, value: Bytes) -> crate::Result<()> {
+        self.rt.block_on(self.inner.set(key, value))
+    }
+
+    pub fn set_expires(
+        &mut self,
+        key: &str,
+        value: Bytes,
+        expiration: Duration,
+    ) -> crate::Result<()> {
+        self.rt.block_on(self.inner.set_expires(key, value, expiration))
+    }
+
+    pub fn publish(&mut self, channel: &str, message: Bytes) -> crate::Result<u64> {
+        self.rt.block_on(self.inner.publish(channel, message))
+    }
+}
+```
+The [`Client::subscribe`] method is more interesting because it transforms the
+`Client` into a `Subscriber` object. We can implement it in the following
+manner:
+```rs
+/// A client that has entered pub/sub mode.
+///
+/// Once clients subscribe to a channel, they may only perform pub/sub related
+/// commands. The `Client` type is transitioned to a `Subscriber` type in order
+/// to prevent non-pub/sub methods from being called.
+pub struct Subscriber {
+    /// The asynchronous `Subscriber`.
+    inner: crate::client::Subscriber,
+
+    /// A `current_thread` runtime for executing operations on the asynchronous
+    /// `Subscriber` in a blocking manner.
+    rt: Runtime,
+}
+
+impl Client {
+    pub fn subscribe(self, channels: Vec<String>) -> crate::Result<Subscriber> {
+        let subscriber = self.rt.block_on(self.inner.subscribe(channels))?;
+        Ok(Subscriber {
+            inner: subscriber,
+            rt: self.rt,
+        })
+    }
+}
+
+impl Subscriber {
+    pub fn get_subscribed(&self) -> &[String] {
+        self.inner.get_subscribed()
+    }
+
+    pub fn next_message(&mut self) -> crate::Result<Option<Message>> {
+        self.rt.block_on(self.inner.next_message())
+    }
+
+    pub fn subscribe(&mut self, channels: &[String]) -> crate::Result<()> {
+        self.rt.block_on(self.inner.subscribe(channels))
+    }
+
+    pub fn unsubscribe(&mut self, channels: &[String]) -> crate::Result<()> {
+        self.rt.block_on(self.inner.unsubscribe(channels))
+    }
+}
+```
+So, the `subscribe` method will first use the runtime to transform the
+asynchronous `Client` into an asynchronous `Subscriber`. Then, it will store the
+resulting `Subscriber` together with the `Runtime` and implement the various
+methods using [`block_on`].
+
+Note that the asynchronous `Subscriber` struct has a non-async method called
+`get_subscribed`. To handle this, we simply call it directly without involving
+the runtime.
+
+# Other approaches
```

Comment:
The rest of the page is fine. I think it would be more effective if the examples were in context of mini-redis, e.g. showing how these strategies could work for a similar problem. 

For example, instead of `my_bg_task`, it could be a `get` request, which would enable the caller to run multiple parallel redis GET requests from a blocking context.